guide_drafts/style_guide.md

```diff
@@ -0,0 +1,171 @@
+
+# Diesel Style Guide
+
+## Philosophy
+
+The goal of these style guidelines is to maintain a useful git history,
+particularly the ability to get context on a line of code quickly via `git blame`.
+As such, much of the code is styled in such a way that lines are unlikely to change
+in order to support unrelated changes. 
+
+Note that these are guidelines, not hard rules. Use your best judgement.
+
+**If you find existing code that doesn't already adhere to these guidelines, don't change it simply to adhere to them.**
+
+If we start changing code for style reasons only, it completely defeats the purpose of trying to avoid git churn in the first place. Do change code that doesn't adhere to these guidelines if you're already changing that line for other reasons.
+
+## Guidelines
+
+### Brakets on their own lines
+
+If a function, struct, trait, or any other definition spans more than one line, the `{` goes on its own line.
+
+Example:
+
+``` rust
+// GOOD
+fn foo(some_args: OmgVerboseTypeName, more_args: OmgVerboseTypeTwo)
+    -> OurReturnType
+{
+    // ...
+}
+
+// BAD
+fn foo(some_args: OmgVerboseTypeName, more_args: OmgVerboseTypeTwo)
+    -> OurReturnType {
+    // ...
+}
+
+// ALSO BAD
+fn foo(some_args: OmgVerboseTypeName, more_args: OmgVerboseTypeTwo)
+    -> OurReturnType {
+        // ...
+    }
+```
+
+Why?
+While it costs vertical space,
+the opening bracket visually separates the function signature and the fuction body
+which may be at the same indent level (as above).
+Additionally, we need the indentation level of the function body to be independent of the struct definition.
+
+### Do not token align
+
+The next line should start
+4 spaces deeper than the previous,
+at the same indentation level,
+or 4 spaces shallower,
+not 7 because it lets a `:` line up.
+
+Examples:
+
+``` rust
+// GOOD
+fn my_function(
+    this_is_a_longer_arg: Type1,
+    shorter_arg: Type2,
+) -> ReturnType {
+{
+    // ...
+}
+
+// BAD
+fn my_function(
+    this_is_a_longer_arg: Type1,
+             shorter_arg: Type2,
+) -> ReturnType {
+    // ...
+}
+```
+
+Why? It rarely improves readability
+and it pollutes the git history
+as we'll have to re-align everything as soon as we add a longer key.
+
+
+### Always use trailing commas for multiline groupings
+
+Example:
+
+``` rust
+// GOOD
+let ints = [
+    1,
+    2,
+    3,
+]
+
+// BAD
+let ints = [
+    1,
+    2,
+    3
+]
+```
+
+### Keep lines under ~80 characters
+
+For code, this is a soft guideline. You don't need to break it up when it's 81.
+A rough guideline is \"Does this fit in my terminal pane?\",
+which for most of us comes out to roughly 80ish characters,
+but there's no _hard_ character limit.
+
+Documentation should have a _hard_ wrap at 80 characters.
```

Comment:
Hard wrap at 80 cols _in total_ or hard wrap at "indentation + `/// ` + 80 cols of text"? I think rustfmt uses the latter (or actually the minimum of the latter and 100 cols)